libs/agno/agno/os/client.py

```diff
@@ -0,0 +1,2121 @@
+            session_id: Optional session ID for context
+            user_id: Optional user ID
+            images: Optional list of Image objects
+            audio: Optional list of Audio objects
+            videos: Optional list of Video objects
+            files: Optional list of MediaFile objects
+            **kwargs: Additional parameters (session_state, dependencies, metadata, etc.)
+
+        Yields:
+            RunOutputEvent: Typed event objects (RunStartedEvent, RunContentEvent, etc.)
+
+        Raises:
+            HTTPStatusError: On HTTP errors
+        """
+        endpoint = f"/agents/{agent_id}/runs"
+        data: Dict[str, Any] = {"message": message, "stream": "true"}
+        if session_id:
+            data["session_id"] = session_id
+        if user_id:
+            data["user_id"] = user_id
+        if images:
+            data["images"] = json.dumps([img.model_dump() for img in images])
+        if audio:
+            data["audio"] = json.dumps([a.model_dump() for a in audio])
+        if videos:
+            data["videos"] = json.dumps([v.model_dump() for v in videos])
+        if files:
+            data["files"] = json.dumps([f.model_dump() for f in files])
+
+        for key, value in kwargs.items():
+            if isinstance(value, dict):
+                data[key] = json.dumps(value)
+            else:
+                data[key] = value
+
+        # Get raw SSE stream and parse into typed events
+        raw_stream = self._stream_post_form_data(endpoint, data)
+        async for event in self._parse_sse_events(raw_stream, run_output_event_from_dict):
+            yield event
+
+    async def continue_agent_run(
+        self,
+        agent_id: str,
+        run_id: str,
+        tools: List[Dict[str, Any]],
+        stream: bool = False,
+        session_id: Optional[str] = None,
+        user_id: Optional[str] = None,
+    ) -> RunOutput:
+        """Continue a paused agent run with tool results.
+
+        Args:
+            agent_id: ID of the agent
+            run_id: ID of the run to continue
+            tools: List of tool results to provide
+            stream: Whether to stream the response
+            session_id: Optional session ID
+            user_id: Optional user ID
+
+        Returns:
+            RunOutput: The continued run response
+
+        Raises:
+            HTTPStatusError: On HTTP errors
+        """
+        endpoint = f"/agents/{agent_id}/runs/{run_id}/continue"
+        data: Dict[str, Any] = {"tools": json.dumps(tools), "stream": str(stream).lower()}
+        if session_id:
+            data["session_id"] = session_id
+        if user_id:
+            data["user_id"] = user_id
+
+        response_data = await self._post(endpoint, data, as_form=True)
+        return RunOutput.from_dict(response_data)
+
+    async def stream_continue_agent_run(
+        self,
+        agent_id: str,
+        run_id: str,
+        tools: List[Dict[str, Any]],
```

Comment:
I don't think this type would be right. It would be a list of ToolExecutions I think?